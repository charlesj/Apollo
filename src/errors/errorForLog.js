import serializeError from 'serialize-error'

export function errorForLog(obj) {
  if(obj instanceof Array){
    return obj.reduce((acc, val, index) => {
      acc[index] = serializeError(val)
      return acc
    }, {})
  }
  return serializeError(obj)
}
