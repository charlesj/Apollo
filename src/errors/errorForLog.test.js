
import { errorForLog } from './errorForLog'

describe('errorForLog', () => {
  it('converts a top level error', () => {
    const error = new Error('Top level error')
    const cleaned = errorForLog(error)
    expect(Object.keys(cleaned)).toMatchSnapshot()
  })

  it('converts a nested error', () => {
    const error = new Error('Top level error')
    const cleaned = errorForLog({ error })
    expect(Object.keys(cleaned.error)).toMatchSnapshot()
  })

  it('converts an error with an error', () => {
    const error = new Error('Top level error')
    const cleaned = errorForLog([{ error }])
    expect(Object.keys(cleaned[0].error)).toMatchSnapshot()
  })

  it('converts an array into an object', () => {
    const cleaned = errorForLog(['one', { nested: true }])
    expect(cleaned).toMatchSnapshot()
  })
})
