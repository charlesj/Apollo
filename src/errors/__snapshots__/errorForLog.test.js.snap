// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorForLog converts a nested error 1`] = `
Array [
  "name",
  "message",
  "stack",
]
`;

exports[`errorForLog converts a top level error 1`] = `
Array [
  "name",
  "message",
  "stack",
]
`;

exports[`errorForLog converts an array into an object 1`] = `
Object {
  "0": "one",
  "1": Object {
    "nested": true,
  },
}
`;

exports[`errorForLog converts an error with an error 1`] = `
Array [
  "name",
  "message",
  "stack",
]
`;
